### PR TITLE
Revert "Lowers portable pump max pressure to 1013.25"

### PR DESCRIPTION
**Pull request by CPTANT**
Reverts tgstation/tgstation#27894

Merging your own Pr's isn't allowed. Letting someone else change that one number to subsequently merge it yourself is just blatantly trying to fuck over that rule.

Nobody liked this change.

### CODE REVIEW

**Comment by kevinz000** [comment]
1. Acquire proficiency
2. Didn't know shadowlight is also leoz!

**Comment by CPTANT** [comment]
> Requested by @lzimann 
> 
>  lzimann merged commit fcde4be into tgstation:master

**Comment by Haevacht** [comment]
Hi Nobody, this is a good PR you've made here.

**Comment by octareenroon91** [comment]
1. The topic of this PR could have used more discussion before a merge rather close to the 24-hour mark.
2. This PR is turning into a shitstorm, unfortunately. We've entered policy- rather than code-problem territory.

**Comment by CPTANT** [comment]
Yes and personally I think merging your own changes after 24hours with 82% percent of people disliking the change is something that shouldn't happen.

Also it was already argued while this was a useless change. It does nothing to stop the problem it claims to fix and reduced functionality for other uses.

**Comment by Grazyn** [comment]
@CPTANT the claim was that making 22/44/88 bombs was too easy, and now it is fixed. I don't think it was needed because lavaland nuking or traitors tc trading for cap removal wasn't something that happened so often but that was the scope of the PR, not bomb making itself.

Now, If plebs somehow aquire the skill needed to make nukes without 3k pumps then a new PR will be needed, and eventually bomb making will be changed again, and the cycle will begin anew.

**Comment by LaharlMontogmmery** [comment]
From an external PoV : 

1. The base premise of the original PR sounds elitist, but it has grounds to stand on - as in making 20-dev bombs should require a little more work then the air pumps. It is merely closing one path to bombing greatness, however, and it might discourage normal scientists from creating/handing bombs to mining, because it requires more effort than any other departments. But I don't think this is the problem here - this was merely a balance issue.
2. Lzi's merging. As it turns out, the change was requested by Lzi himself, he was the main defendant of the PR and then was also merged by him. Now, there's two things that might have happened 

- Lzi asked about it on the bus, there was a consensus between maintainers, and Lzi merged it. A bit too hasty, and he should have let another maintainer take a look at it and merge it, but okay, sure, **that's fine, even if a bit iffy-looking.**

- There was no consensus, and Lzi merged it as soon as the 24h period was over to get it over with. This is where things start not looking good : why? Because that means that he acted as both **demander, defendant and judge, while having a huge bias towards merging the PR.** That means he essentially forced the PR through without any inputs from other maintainers. **This isn't OK, and it is abusing the current system.**

TLDR : @lzimann should have probably let another maintainer take a look at the PR and merge it, instead of strong-arming it - it would have gone through, I think. Instead, we're going to have a policy shitstorm.

**Comment by optimumtact** [comment]
maintainers don't need to get consensus

**Comment by LaharlMontogmmery** [comment]
@optimumtact If that's the case, Lzi merged a balance change he himself requested, with a quite apparent bias.
I'm not sure if that's the way things are supposed to go for balance changes.

**Comment by Grazyn** [comment]
You should just remove the clause that maintainers aren't allowed to merge their own PRs tbh.

Shadowlight didn't have anything to do with the PR, when I ahelped him in-game to tell him that the change had no impact whatsoever on maxcap-making he told me that he was hands-off with the PR and to just write it in the comments. It was obvious that the change was requested by Lzimann, being the only coder/player who cares and knows enough about atmos autism. It was also obvious that the PR was going to be merged no matter what.

Now if Shadowlight had put "Huge bombs are too easy to make and erasing 1/4th of lavaland in the first 5 minutes of the round defeats the whole point of mining" in the description this wouldn't have been a problem, it's just the blatant disregard for the clause ("Duh, here's a nerf, I don't know, it was requested") that it's bound to cause drama, expecially if it becomes common practice. Remove the clause and everyone will be happy.

**Comment by RemieRichards** [comment]
The clause is there so that maintainer PRs actually get reviewed, same as others.
Yes we're held to a higher standard in terms of code quality, because that's why were maintainers, we've already shown we're capable.
But mistakes happen, and the more eyes on something the better, plus if you've spent say an hour or two working on a PR your mind might be very clear on what something does or how it works, but it might not even be readable to anyone else.

tl;dr it's valuable, and we try not to break it, unlike some of the other rules that are loose (like 24hrs)

**Comment by Grazyn** [comment]
@RemieRichards That wouldn't be a problem if a maintainer's decision to merge was based on code quality alone, but it's my understanding that maintainers also decide on grounds of balance, gameplay, powercreep and so on. The 2-man rule guarantees that at least another maintainer reviews it and approves it on those grounds. That's why someone can't make a PR that gives deathsquad armor to the mime and expect it to be merged. 

This whole thing is defeated when a maintainer asks another coder (who may not even understand or care about what's he's being asked to do) to change something, and then reviews and merges it himself. This isn't even about this specific case since I'm sure the PR would've been merged by another maintainer anyway.

**Comment by RemieRichards** [comment]
You're right that making a change via proxy does circumvent the rule, but it's no reason to remove the rule.

**Comment by lzimann** [comment]
I didnt even specifically requested the change. I mentioned about it when we were discussing bombs one day, which ended up in shadow making the pr. I was probably going to do it eventually, and guess what, it would've been merged regardless.

**Comment by LaharlMontogmmery** [comment]
@lzimann I get what you mean, there was a good chance it would have been merged regardless - that's not the point, tho. I'll better illustrate with an example of server policy :
 
As admins, we have a policy where we don't handle adminhelps where we have been implicated IC.
I feel like the same should be applied to maintainers, AKA they should not be resolving/merging PR where they have been implicated (meaning here : requesting the change or being a major part in the PR debate).

I'm insisting on that point, because it seems to me it can lead to more abuse of our current system.

**Comment by duncathan** [comment]
>It was obvious that the change was requested by Lzimann, being the only coder/player who cares and knows enough about atmos autism

now that just hurts

**Comment by duncathan** [comment]
I was gonna merge that PR anyway, btw. It's a good change. 10 atmospheres should be the output limit on portable atmos.

**Comment by duncathan** [comment]
This conversation is still worth having, but the PR itself was gonna go through regardless.

**Comment by evsey9** [comment]
It's impossible to get any tank to 3039 kPa now.

**Comment by duncathan** [comment]
That is objectively untrue. If pressure is your goal, you can easily just heat the tank to increase pressure. If you're concerned with actual amount of gas, you're better off space cooling some gas and putting it in at 10 atmospheres than to just throw in 30 atmospheres at room temperature.
If you really need 30 atmospheres of gas for some reason, there are other slightly more complicated ways to achieve it. Ask lzi, he's more well versed in the practical side of atmos autism.

**Comment by evsey9** [comment]
wait, you can individually heat tanks??

**Comment by Grazyn** [comment]
I don't know if jetpacks work off moles or pressure, but it's perfectly possible to get a jetpack with >2000 kPa of oxygen, it won't be supercooled though

**Comment by lzimann** [comment]
Jetpacks are just tanks with some extra things to let you move in space/no grav.

**Comment by Lumipharon** [comment]
The pr makes no sense.

If making nukes for lavaland is an issue, then NERF THE ABILITY TO NUKE LAVALAND.
The pr was literally 'why should the plebs be able to do what I can do, reee". It does not at all fix the issue of being able to nuke the shit out of lavaland, it just reduces the number of people that know how to do it.

If someone just takes 5 minutes to type up a guide to making nukes without air pumps and sticks it back on the wiki, so everyone can do it again, will everyone be ok with that? Because if not, THAT PR WAS POINTLESS, and if you think it is ok, THEN THAT PR WAS STILL POINTLESS.

How is this valid by any stretch of the imagination?

**Comment by lzimann** [comment]
Then the cycle of bomb tweaking continues.

**Comment by VeryRealAccount** [comment]
It's pure elder secrets bullshit that Izimann ignored the rules to get through.

**Comment by optimumtact** [comment]
welcome to the github

**Comment by Grazyn** [comment]
Project is open source and toxins is literally the only thing that you can't understand and master just by looking at the code. Of course veterans are gonna protect it from the plebs. 
You can still make powerful bombs if you play with heat and pressure, even if you don't know elder secrets.

0345982ff12c62a3c705f8c6b2b20d05

**Comment by lcass** [comment]
You can still make powerful bombs but maybe instead of just removing the tools to make them you should change the core bomb code to favour people using mixes instead of just a pump? 
Atleast consider putting the limit to 20 * atmos pressure so that we can actually see some decent bombs instead of them reaching a maximum at 19/38/76.

**Comment by lzimann** [comment]
10 atmospherics is more than enough.

**Comment by lcass** [comment]
You haven't given any reasoning behind this other than that you want your bombs to be obtained by different means.

**Comment by CPTANT** [comment]
This change nerfs flamethrowers more than it does bombs
it nerfs power fists more than it does bombs
it nerfs jetpacks more than it does bombs
it nerfs collectors more than it does bombs

So this nerfs everything except for the stated goal.

**Comment by Lumipharon** [comment]
^

**Comment by Cheridan** [comment]
I'm reopening this based on the circumstances around the merge and the relative unpopularity of the original PR.

**Comment by lzimann** [comment]
A 1 click button that safely makes your tank full(max pressure) is stupid, it's one of those "QoL" changes, removing any risk of tank leak when you try to get a decent pressure. As I showed in the last pr, you can still get tanks with this pressure without a portable pump.

**Comment by Lumipharon** [comment]
>QOL changes

When was it not 3k pressure from air pumps?
And what is the issue with being able to properly fill a tank with a specialsed device, other than 'nukes on lavaland'?

**Comment by Grazyn** [comment]
I'm conflicted about seeing this PR re-opened, even though I was originally against Lzimann's PR (it appeared to be a "merge-by-proxy" and it also broke this other rule when lzimann closed this revert https://tgstation13.org/phpBB/viewtopic.php?f=5&t=10933).

The reason why I'm conflicted is because I see 2 outcomes if pumps are reverted to the original 3k pressure, the most likely one being Lzimann pushing a new refactor of bombs which will lock even regular maxcaps above most players' skill ceiling (and endgame freezer upgrades), until the "perfect recipe" is leaked, and the cycle begins anew until bombs are finally removed. As I said before, maxcaps aren't affected by the new pump limit (as far as skill and established procedures are concerned), only nukes are.

The second, less likely outcome is a lower bombcap on lavaland, this will disappoint everyone who likes to push the boundaries of the cap and wants to see the result in practice and not just as a line from the tachyon array. Yes, traitors can TC trade for cap removal but I've only seen it happen once in the last 6 months, the traitor was Robustin and the admin was Kevinz, other admins won't let you do it because it's literally only fun for the traitor.

Also, about 3k pressure pumps affecting other things: I don't play engineering but I think that collectors filled with supercooled plasma tanks at 3k kPa would yield enormous amounts of power regardless of the supermatter gas mix you use. And a 3k kPa jetpack gives you 3 times the fuel distance with minimum effort.

**Comment by lcass** [comment]
I think the point here is then to not let lziman change the code because they want maxcaps to be secret that only they know. If they are just going to debuff code they don't like they shouldn't be allowed on the coding team as they aren't doing what's in the interest of the playerbase.

It takes a lot of time to make a proper bomb mix and having it changed every single time is insanely annoying as the mixes are already pretty secretive. In my entire time playing toxins I have only met 3 people who could make 22/44/88 bombs and maybe about 6 - 7 more who could go up to > 15.

If you are going to argue that too many people know how to just win it then you need to change R&D as well because if anything that is known by basically everyone and can be completed with resources in about 10 minutes,

**Comment by Shadowlight213** [comment]
Personally, I care less about the bomb situation now, and more of agree with dunc that having something that's supposed to be a portable pump, having the highest pressure setting ability for filling tanks, as opposed to some sort of machine in atmospherics is silly.

**Comment by Lumipharon** [comment]
I unno man, google scuba tanks and air compressors, if anthing, the max pressure in ss13 is actually low.
Unless I am misreading shit, lots of these tanks can fill up to like 30,000 kpa, and air compressors are pretty simple machines, so realism is not an argument here.

**Comment by CPTANT** [comment]
Also the stationary pumps have higher pressure output to begin with. There just isn't anything that fills portable tanks.

**Comment by octareenroon91** [comment]
>Also the stationary pumps have higher pressure output to begin with. There just isn't anything that fills portable tanks.

Too true. The highest commonly-seen pressure is 4500 kPa, with the atmos storage lines going up to 5066. There's no option for direct pipeline to tank connection though.

**Comment by lzimann** [comment]
I might get a lot of hate for closing, but 4 days and no one even bothering with it, so, let's clear the dirty tracker, shall we?

**Comment by Lumipharon** [comment]
Wew. Lad.

**Comment by lzimann** [comment]
L, it i

**Comment by PKPenguin321** [comment]
Violated the revert rule, https://tgstation13.org/phpBB/viewtopic.php?f=5&t=10933#p287839

**Comment by lzimann** [comment]
It is not written anywhere nor no one told me. I can reopen and let it sit for another 30 days if you want.

**Comment by PKPenguin321** [comment]
Yeah

**Comment by lcass** [comment]
It didn't sit for 30 days there is a discussion going on but people are getting bored because the same points are being made and no conclusion is being reached.

**Comment by Lumipharon** [comment]
Because my points go un answered.
What is the purpose of nerfing the air pump other than making it harder to make big bombs that are only usable on lavaland? 
Because if there aren't good reasons, the change makes no sense because it doesn't fix the issue in question (nuking lava land), it just means it's only abusable by the person pushing the change and a handful of others.

**Comment by CPTANT** [comment]
I want it reopened and someone who is not Izimann to make a decision.

**Comment by optimumtact** [comment]
you can't have nice things

**Comment by CPTANT** [comment]
guess nobody gave a shit, despite Izimann closing his own revert again.

**Comment by optimumtact** [comment]
nobody cares okay
